python/tvm/micro/base.py

```diff
@@ -0,0 +1,131 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Base definitions for micro."""
+
+from __future__ import absolute_import
+
+import logging
+import os
+
+from tvm.contrib import util
+
+from .._ffi.function import _init_api
+from .._ffi.libinfo import find_include_path
+from .cross_compile import create_lib
+
+SUPPORTED_DEVICE_TYPES = ["host"]
+
+class Session:
+    """MicroTVM Session
+
+    Example
+    --------
+    .. code-block:: python
+
+      c_mod = ...  # some module generated with "c" as the target
+      device_type = "host"
+      with tvm.micro.Session(device_type) as sess:
+          sess.create_micro_mod(c_mod)
+    """
+
+    def __init__(self, device_type, toolchain_prefix):
+        """Stores parameters for initializing a micro device session.
+
+        The session is not initialized until the constructed object is used
+        in a `with` block.
+
+        Parameters
+        ----------
+        device_type : str
+            type of low-level device
+
+        toolchain_prefix : str
+            toolchain prefix to be used. For example, a prefix of
+            "riscv64-unknown-elf-" means "riscv64-unknown-elf-gcc" is used as
+            the compiler and "riscv64-unknown-elf-ld" is used as the linker,
+            etc.
+        """
+        if device_type not in SUPPORTED_DEVICE_TYPES:
+            raise RuntimeError("unknown micro device type \"{}\"".format(device_type))
+
+        # First, find and compile runtime library.
+        micro_dir = os.path.dirname(os.path.realpath(os.path.expanduser(__file__)))
+        micro_device_dir = os.path.join(micro_dir, "..", "..", "..",
+                                        "src", "runtime", "micro", "device")
+        runtime_src_path = os.path.join(micro_device_dir, "utvm_runtime.c")
+        tmp_dir = util.tempdir()
```

Comment:
NOTE: tmp_dir will get deleted when it goes out of scope. If you want to keep it alive, you need to keep it as a class member. This could be the reason why your test failed.

Reply:
I don't think this is the problem.  The `_CreateSession` call below reads from `tmp_dir` and then `tmp_dir` is no longer needed.